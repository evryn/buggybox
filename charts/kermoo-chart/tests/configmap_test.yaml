suite: test configmap
templates:
  - configmap.yaml
tests:
  -
    it: manifest with default values should match snapshot
    asserts:
      - matchSnapshot: {}
  -
    it: works with extra common labels
    set:
      commonLabels:
        my-label: hello
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: metadata.labels["my-label"]
          value: hello
  -
    it: works with extra common annotations
    set:
      commonAnnotations:
        my-annotation: hello
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: metadata.annotations["my-annotation"]
          value: hello
  -
    it: works with custom configuration
    set:
      configuration: |
        hello:
          world: ok
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: data["config.yaml"]
          value: |
            hello:
              world: ok
