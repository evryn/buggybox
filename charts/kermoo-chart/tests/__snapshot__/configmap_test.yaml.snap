manifest with default values should match snapshot:
  1: |
    apiVersion: v1
    data:
      config.yaml: |2
        process:
          delay: 1s
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-config
